lib/search_terms/keyword.rb

```diff
@@ -1,8 +1,14 @@
 class Keyword
-  def initialize(string)
+  def initialize(string,mode = :sensitive)
```

Comment:
There are two ways we can do the constructor differently. The "classic" approach is with an options hash. If the last argument to a method (any method--not just a constructor) is hash-syntax then Ruby will collect the everything up and create a hash for you. This works in all versions of Ruby in use today:

```ruby
class Keyword
  def initialize(string, opts = {})
    @mode = opts.fetch(:mode, :sensitive) # use fetch to set the default when no mode given
    #@sensitive = opts.fetch(:sensitive, true) # alternatively, make it a boolean
    ...
  end
  ...
end

foo = Keyword.new('foo') # mode will be :sensitive
bar = Keyword.new('bar', mode: :insensitive) # mode will be :insensitive
```

Starting with Ruby 2.0 (not yet supported in JRuby) we have keyword arguments. The syntax is designed to look like the classic opts hash but *not* actually use an opts hash (certain edge cases don't play nicely with the class opts hash approach, such as variable-length arguments and arguments with defaults). The behavior of Ruby's keyword arguments is based on a similar feature in Python.

```ruby
class Keyword
  def initialize(string, mode: :sensitive)
    # NOTE: a keyword argument *must* have a default value, which can be any data type
    @mode = mode
    ...
  end
  ...
end

# notice that these two lines are identical to the first example--backward compatibility FTW
foo = Keyword.new('foo') # mode will be :sensitive
bar = Keyword.new('bar', mode: :insensitive) # mode will be :insensitive
```